File: java_solutions/1238.circular-permutation-in-binary-representation.java
/*
 * @lc app=leetcode id=1238 lang=java
 *
 * [1238] Circular Permutation in Binary Representation
 */

class Solution {
    public java.util.List<Integer> circularPermutation(int n, int start) {
        return java.util.stream.IntStream.range(0, 1 << n).mapToObj(i -> start ^ i ^ (i >> 1)).collect(java.util.stream.Collectors.toList());
    }
}

File: kotlin_solutions_1/1238.circular-permutation-in-binary-representation.kt
/*
 * @lc app=leetcode id=1238 lang=kotlin
 *
 * [1238] Circular Permutation in Binary Representation
 */
class Solution { fun circularPermutation(n: Int, start: Int): List<Int> = (0 until (1 shl n)).map { i -> start xor i xor (i shr 1) } }